experiment/logviewer/gcsreader/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+)
+
+func main() {
+	defer timeTrack(time.Now(), "total run")
+	defer durations.printAll()
+	run()
+}
+
+func run() {
+	logPath, targetSubstring := setupFromConsole()
```

Comment:
I think we should separate line like:
func grep(r io.Reader) []logEntry

Then we can prepare io.Reader based on the parameters (this can happen in main) and call grep on opened io.Reader.

This way we will be able to easily test this function and in fact this is the only function here that I think we should test.

Reply:
@mborsz could you please, clarify, what do you mean by this `grep` function? What is the difference between this function and current [processLines](https://github.com/kubernetes/test-infra/pull/11488/files/2de65c2e5097e7c040e78d9171fc1db7ddd180d6#diff-9b37afbd8db26b18044ba556b243f810R56) function?

Reply:
Sorry, my fault -- processLines does exactly what I meant.